### DISCUSSION TITLE
Receiving FSK sent over SSB

### DISCUSSION THREAD

**Discussion by abishopgylys** [jgromes/RadioLib#433] [Q&A]
Hi,

I'm new to radio in general and am still figuring out many things. I read online that one can produce FSK using a SSB transmitter by feeding audio tones into the mic port. Is this true, and can the boards supported by RadioLib receive FSK produced in this manner?

If not, are there any computer-controllable transmitters that can produce plain FSK receivable by RadioLib-supported boards? All the computer-controllable transmitters I've come across seem to only support RTTY FSK, which I read is not receivable by RadioLib-supported boards. Alternatively, is there a way to modify or use an RTTY FSK transmitter to transmit plain FSK?

Thanks!

**Comment by jgromes** [comment]
> I read online that one can produce FSK using a SSB transmitter by feeding audio tones into the mic port

Unfortunately I'm not the best person to ask this sort of question, sorry. What you're describing seems reasonable. However, I suspect this would only be suitable to generate slow FSK, 1200 bps or slower. Most modules supported by RadioLib can't receive FSK that slow, the typical speeds are 10 - 100 kbps.

> are there any computer-controllable transmitters that can produce plain FSK receivable by RadioLib-supported boards

One could argue that any Arduino with one of the modules supported by RadioLib fits that description quite well ;) Although I suspect that is not exactly what you're looking for. Other than that, I'm not aware of any, sorry.

> RTTY FSK, which I read is not receivable by RadioLib-supported boards

That's correct, the radio modules here are designed as digital packet transceivers - all the ham radio modes supported by RadioLib are basically hacks, and definitely not what the modules were intended to do. Unfortunately receiving RTTY with these modules is not technically possible as far as I can tell.

**Reply by abishopgylys** [reply]
Thanks so much for the quick response!

When you say "most" modules can't receive it that slow, does that include the RFM96W?

And haha yes, I'll probably end up taking the Arduino route on this one - just wanted to use a 'traditional' transmitter if I could.

**Reply by jgromes** [reply]
The absolute slowest you can go on RFM96 is 1200 bps, which might be slow enough. In addition, the lowest frequency deviation that can be set is 600 Hz, so in theory, a 1200/2400 Hz tones at 1200 baud **could** work. That puts the module at 1.2 kbps and deviation of 1.2 kHz. I have no idea if this will work in practice though.

You will have to construct the packet structure in the audio part, so you will have to synthesize a preamble and preferrably a sync word and data length, that's the default packet format. Alternatively, you could use direct reception (that doesn't use packets at all).

To sum it up, this might be possible, but seems overly complicated. I definitely wouldn't start with this, since you will have to be familiar with the basic concepts like the packet structure. I would suggest getting two RFM69 modules with two Arduino boards, getting them to talk together, examine the signal with SDR etc. just to get familiar with them.
